operator/v1alpha1/types_kueue.go

```diff
@@ -0,0 +1,154 @@
+	// items is a slice of kueue
+	// this is a cluster scoped resource and there can only be 1 kueue
+	// +kubebuilder:validation:MaxItems=1
+	// +required
+	Items []Kueue `json:"items"`
+}
+
+// +kubebuilder:validation:Enum=batchjob;rayjob;raycluster;jobset;mpijob;paddlejob;pytorchjob;tfjob;xgboostjob;appwrapper;pod;deployment;statefulset;leaderworkerset
+type KueueIntegrations string
+
+const (
+	BatchJob        KueueIntegrations = "batchjob"
+	RayJob          KueueIntegrations = "rayjob"
+	RayCluster      KueueIntegrations = "raycluster"
+	JobSet          KueueIntegrations = "jobset"
+	MPIJob          KueueIntegrations = "mpijob"
+	PaddeJob        KueueIntegrations = "paddlejob"
+	PyTorchJob      KueueIntegrations = "pytorchjob"
+	TfJob           KueueIntegrations = "tfjob"
+	XGBoostJob      KueueIntegrations = "xgboostjob"
+	AppWrappers     KueueIntegrations = "appwrapper"
+	Pod             KueueIntegrations = "pod"
+	Deployment      KueueIntegrations = "deployment"
+	Statefulset     KueueIntegrations = "statefulset"
+	LeaderWorkerSet KueueIntegrations = "leaderworkerset"
+)
+
+// This is the GVK for an external framework.
+// Controller runtime requires this in this format
+// for api discoverability.
+type ExternalFramework struct {
+	// group of externalFramework
+	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:XValidation:rule="self.size() == 0 || !format.qualifiedName().validate(self).hasValue()",message="a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character."
+	// +required
+	Group string `json:"group"`
+	// resource of external framework
+	// +required
+	// +kubebuilder:validation:MaxLength=256
+	// +kubebuilder:validation:XValidation:rule="self.size() == 0 || !format.qualifiedName().validate(self).hasValue()",message="a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character."
+	// +kubebuilder:validation:MinLength=1
+	Resource string `json:"resource"`
+	// version is the version of the api
+	// +required
+	// +kubebuilder:validation:MaxLength=256
+	// +kubebuilder:validation:MinLength=1
+	Version string `json:"version"`
+}
+
+type Integrations struct {
+	// frameworks are a list of names to be enabled.
+	// This is required and must have at least one element.
+	// The frameworks are jobs that Kueue will manage.
+	// +kubebuilder:validation:MaxItems=14
+	// +kubebuilder:validation:MinItems=1
+	// kubebuilder:validation:UniqueItems=true
+	// +listMapKey=atomic
+	// +required
+	Frameworks []KueueIntegrations `json:"frameworks"`
+	// externalFrameworks are a list of GroupVersionResources
+	// that are managed for Kueue by external controllers;
+	// These are optional and should only be used if you have an external controller
+	// that integrations with kueue.
```

Comment:
You're mixing the casing of `kueue` quite a lot through the godoc, I think it should be capitalised as a noun
```suggestion
	// that integrates with Kueue.
```